parser/Builder.cc

```diff
@@ -832,6 +885,13 @@ class Builder::Impl {
         return make_unique<NthRef>(tokLoc(tok), atoi(tok->string().c_str()));
     }
 
+    unique_ptr<Node> numparams(size_t max_numparam) {
+        // At this point, we might not have a node to get the location so we use a dummy one for now,
+        // errors will later be rattached to the block itself.
```

Comment:
Why will the errors be there? Is it because desugar removes this node entirely?

Reply:
It's because of the way we parse them: we discover the parsing of the block, the block body is create before the block params. This means that when we create the numparam node, the location of the block params is not yet available (the block params production will be constructed after the block body one). We could attach it to the location of the block body but the body itself might be an empty tree without any token so no location either.

Instead of playing with 3 different `maybeLoc`, I opted to always use the one of the block during desugar.

Reply:
Forget my previous message. I think we can use the `numparam_stack` to store the locations as we find them and provide better errors messages.

Reply:
Done